.github/workflows/continuous-integration.yaml

```diff
@@ -7,21 +7,106 @@ on:
   pull_request:
     branches:
     - master
+  schedule:
+    - cron: '0 2 * * 1' # At 02:00 on Monday
+
+permissions: {}
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 15
     strategy:
       matrix:
-        node-version: [10, 12, 14]
+        os: [ubuntu-latest]
+        node-version: [10, 12, 14, 16]
+        include:
+          - os: macos-latest
+            node_version: 14
+          - os: windows-latest
+            node_version: 14
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Update NPM
+        if: matrix.os != 'macos-latest'
+        run: npm install --global npm
+      - name: Update NPM (macOS)
+        if: matrix.os == 'macos-latest'
+        run: npm install --global --force npm
```

Comment:
This is a workaround for the following error:

```
 npm WARN rm not removing /usr/local/bin/npm as it wasn't installed by /usr/local/lib/node_modules/npm
npm WARN rm not removing /usr/local/bin/npx as it wasn't installed by /usr/local/lib/node_modules/npm
npm ERR! code EEXIST
npm ERR! syscall symlink
npm ERR! path ../lib/node_modules/npm/bin/npm-cli.js
npm ERR! dest /usr/local/bin/npm
npm ERR! errno -17
npm ERR! EEXIST: file already exists, symlink '../lib/node_modules/npm/bin/npm-cli.js' -> '/usr/local/bin/npm'
npm ERR! File exists: /usr/local/bin/npm
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/runner/.npm/_logs/2021-09-26T12_28_41_210Z-debug.log
```

*Taken from: https://github.com/loopbackio/loopback-connector/runs/3712828899?check_suite_focus=true*